src/diff/index.js

```diff
@@ -87,11 +87,11 @@ export function diff(dom, parentDom, newVNode, oldVNode, context, isSvg, excessD
 			c._vnode = newVNode;
 
 			// Invoke getDerivedStateFromProps
-			let s = c._nextState || c.state;
+			if (c._nextState==null) {
+				c._nextState = c.state;
+			}
 			if (newType.getDerivedStateFromProps!=null) {
-				oldState = assign({}, c.state);
-				if (s===c.state) s = c._nextState = assign({}, s);
-				assign(s, newType.getDerivedStateFromProps(newVNode.props, s));
+				c.setState(newType.getDerivedStateFromProps(newVNode.props, c._nextState));
```

Comment:
doesn't this enqueue a double render or is that safeguarded somehow

Reply:
I checked it. maybe no problem.

Reply:
It's likely inert because `component._dirty` is `true`. I'm still a little wary of using setState() here though, it's not an API that was supposed to be used by the diff.

@mochiya98 - any chance you could explain why this works better than setting `c._nextState` directly? Sorry for asking!

Reply:
that is to reduce the bundle size.
I thought it is okay because `setState()` is working when using in lifecycle.
but dependencies will be complicated that i see.

Reply:
Calling `setState` here scares me a bit to be honest. This would introduce a hard dependency in our diff code with the way components are scheduled to render.

Reply:
```jsx
assign(c._nextState===c.state ? (c._nextState = assign({}, c._nextState)) : c._nextState, newType.getDerivedStateFromProps(newVNode.props, s));
```

`setState`: 3440(gz) / 3137(br)
`in diff`: 3462(+22) / 3156(+19)

choose size or dependency...
do you think it is better than?

Reply:
I took that into consideration and decided to write directly.